server/internal/adapter/context.go

```diff
@@ -68,17 +73,16 @@ func Lang(ctx context.Context, lang *language.Tag) string {
 		return lang.String()
 	}
 
-	u := User(ctx)
-	if u == nil {
-		return defaultLang.String()
-	}
-
-	l := u.Lang()
-	if l.IsRoot() {
-		return defaultLang.String()
+	if v := ctx.Value(contextLang); v != nil {
+		if lang, ok := v.(string); ok {
+			if lang == "" {
+				return defaultLang.String()
+			}
+			return lang
+		}
```

Comment:
[memo]
I separated matter of concern about user context from `Lang function`.
This is because, at first glance, the `Lang function` seems to be only interested in contexts related to `lang`.

Therefore, I have used this middleware to get the user's set language and set it in the lang context.

https://github.com/reearth/reearth-visualizer/pull/1342/files#diff-4b6da91f401bf8e39afb161a9a4c2df0acb40e76d910ff79c856e47481bff37c